rollup.config.js

```diff
@@ -0,0 +1,93 @@
+/* eslint-env node */
+'use strict';
+
+const pkg = require('./package.json');
+const commonjs = require('rollup-plugin-commonjs');
+const resolve = require('rollup-plugin-node-resolve');
+const json = require('rollup-plugin-json');
+const babel = require('rollup-plugin-babel');
+const { terser } = require('rollup-plugin-terser');
+
+const DEST_DIR = 'dist';
+const GLOBAL_LIBRARY_NAME = 'purecloud.apps.ClientApp';
+const CJS_FILENAME = 'main.js';
+const ES_FILENAME = 'main.mjs';
+const BROWSER_FILENAME = `${pkg.name}.js`;
+const PROD_BROWSER_EXT = '.min.js';
+
+const LEGAL_TEXT = `/*
+* ${pkg.name}
+* @copyright Copyright (C) ${new Date().getUTCFullYear()} Genesys Telecommunications Laboratories, Inc.}
+* @license MIT
+*
+* This software comprises other FOSS software.
+* Attribution and license information can be found at https://github.com/MyPureCloud/client-app-sdk/blob/master/README.md
+*/`;
+
+const baseRollupConfig = {
+    input: './src/index.js',
+    cache: false,
+    plugins: [
+        commonjs(),
+        resolve(),
+        json(),
+        babel({
+            exclude: 'node_modules/**'
+        })
+    ]
+};
+
+const buildRollupConfig = (config) => {
+    return Object.assign({}, baseRollupConfig, config);
+};
+
+export const umdConfig = buildRollupConfig({
+    output: {
+        dir: DEST_DIR,
+        intro: LEGAL_TEXT,
+        name: GLOBAL_LIBRARY_NAME,
+        entryFileNames: BROWSER_FILENAME,
+        format: 'umd'
+    }
+});
+
+export default [
+    umdConfig,
+    buildRollupConfig({
+        output: {
+            dir: DEST_DIR,
+            intro: LEGAL_TEXT,
+            entryFileNames: ES_FILENAME,
+            format: 'es'
+        }
+    }),
+    buildRollupConfig({
+        output: {
+            dir: DEST_DIR,
+            intro: LEGAL_TEXT,
+            entryFileNames: CJS_FILENAME,
+            format: 'cjs'
+        }
+    }),
+    buildRollupConfig({
+        output: {
+            banner: LEGAL_TEXT,
+            name: GLOBAL_LIBRARY_NAME,
+            dir: DEST_DIR,
+            entryFileNames: `${pkg.name}-[hash]${PROD_BROWSER_EXT}`,
```

Comment:
Is this using internal rollup hashing or rollup-plugin-hash?  I can see the plugin isn't used, but not sure if it's part of rollup proper now?  Looks like the default hash algorithm of the plugin is sha1.

Not that I mind, sha1 is obviously better, but, I noticed that the generated hash length is shorter than the previous md5.  Wondering how it's determining the necessary precision.

Reply:
Yeah, this is now using the template string that's built into rollup https://rollupjs.org/guide/en/#outputentryfilenames. They don't actually say in the docs what algorithm they're using, but I took a quick look at their source code and it seems to be sha256.